Doc/reference/datamodel.rst

```diff
@@ -1613,6 +1613,12 @@ Basic customization
    the left operand's method has priority.  Virtual subclassing is
    not considered.
 
+   When no method returns any value other than NotImplemented, comparisons raise
+   a :exc:`TypeError`. However, the ``==`` and ``!=`` operators will not raise an
+   exception in that case, and will instead fall back to ``is`` and ``is not``,
+   respectively. That way, even though ``object().__eq__(object())`` returns
+   NotImplemented, ``object() == object()`` returns False.
+
```

Comment:
I would prefer a shorter version:
```suggestion
   When no appropriate method returns any value other than ``NotImplemented``, the
   ``==`` and ``!=`` operators will fall back to ``is`` and ``is not``, respectively.
```

Reply:
Ok, but I will add that other comparison operators raise a TypeError, so that no blur remains.

Reply:
I think [it's redundant](https://docs.python.org/3/reference/datamodel.html?#special-method-names): "Except where mentioned, attempts to execute an operation raise an exception when no appropriate method is defined (typically [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) or [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError))."  That's probably why similar cases were not mentioned e.g. in [Emulating numeric types](https://docs.python.org/3/reference/datamodel.html?#emulating-numeric-types).

So, count me -1 on this.

Reply:
Fair enough.